Mod cog debloat

Twentysix26 opened the pull request:
Changes to fix the issues listed in #700

Twentysix26 committed:
[Mod] Moved blacklist / whitelist to owner cog

Twentysix26 committed:
[Mod] Moved [p]modset adminrole / modrole to [p]set

Twentysix26 commented:
So far, this fixes the issue with the server's mod role / admin role not getting checked if the mod cog is disabled.  
I joined `blacklist.json` and `whitelist.json` under a single file in `data/red`. It attempts to import those old files upon creation of the new one, if they exist.

Twentysix26 committed:
[Core] Check mod/admin users against blacklist / whitelist

Users with (server or global) admin or mod role used to be able to skip the whitelist / blacklist checks. This doesn't really make sense from a design pov since these are global checks and the commands are owner only.

Twentysix26 committed:
[Owner] Clearer docstrings and responses

Jan200101 commented:
Currently there is no way of seeing who is whitelisted or blacklisted would love to see that.

Twentysix26 committed:
[Owner] [p]set adminrole/modrole: check for role

Leaves to the user less room for error. If the role doesn't exist it will fail to execute.

Twentysix26 committed:
[Mod] Changed docstrings of old hidden commands

Twentysix26 committed:
[Owner] Added [p]blacklist/whitelist list

Twentysix26 commented:
Good enough for now. I'll have to think about what to do with some other features mentioned in #700.

Merging soon ™️
